core/domain/html_validation_service.py

```diff
@@ -880,6 +880,37 @@ def get_filename_with_dimensions(old_filename, exp_id):
     return new_filename
 
 
+def add_svg_filename_to_math_rte_components(html_string):
+    """Replaces the attribute raw_latex-with-value in all Math RTE tags with
+    a new attribute math_content-with-value. The new attribute has an additional
+    field for storing SVG filenames.
+
+    Args:
+        html_string: str. HTML string to modify.
+
+    Returns:
+        str. Updated HTML string with all Math RTE tags having the new
+        attribute.
+    """
+
+    soup = bs4.BeautifulSoup(
+        html_string.encode(encoding='utf-8'), 'html.parser')
+    for math in soup.findAll(name='oppia-noninteractive-math'):
+        if math.has_attr('raw_latex-with-value'):
+            raw_latex = json.loads(unescape_html(math['raw_latex-with-value']))
+            # Generate an SVG file and filename from the given Latex string.
+            math_expression_contents_dict = {
```

Comment:
Should we generate this using the object class and then normalize/validate it before adding it to the HTML?

Reply:
done